src/Versioning/HasSemVerLikeCompatibility.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Siketyan\Loxcan\Versioning;
+
+use JetBrains\PhpStorm\Pure;
+
+/**
+ * @implements CompatibilityAwareInterface
+ */
+trait HasSemVerLikeCompatibility
+{
+    /**
+     * {@inheritDoc}
+     */
+    #[Pure]
+    public function getCompatibilityNumber(): int
+    {
+        // For 0.Y.* versions, uses Y as a compatibility number instead of X in X.*.* versions.
+        if ($this->getX() === 0) {
+            // To avoid returning the same number from 0.Y.* and X.*.*,
+            // inverts the number to minus here assuming both X and Y are positive.
+            return -$this->getY();
+        }
+
+        return $this->getX() * 10;
```

Comment:
Would you explain more about these parts?

Reply:
Sorry, it was my mistake.
I meant unstable 0.* versions must not have the same "compatibility number" as other stable versions, using negative int.
Now is that clear? [37728e2](https://github.com/siketyan/loxcan/pull/86/commits/37728e273cc74581d4968c2efc3b3c91fb027e7b)

Reply:
@mpyw

Reply:
[e96c256](https://github.com/siketyan/loxcan/pull/86/commits/e96c256203634dc396f78c39af4d15f1708f9b10)